### PR TITLE
Add section for telemetry gathered by the Templating Engine

### PR DESCRIPTION
**Pull request by baronfel**
## Summary

After doing some planning about telemetry for the template engine, I realized that we did in fact collect telemetry from users instantiating templates, and we should declare that in the docs alongside the rest of the declarations.

### CODE REVIEW

File: docs/core/tools/telemetry.md [hunk 1]
```diff
@@ -115,6 +115,13 @@ A subset of commands sends selected options if they're used, along with their va
 
 Except for `--verbosity` and `--sdk-package-version`, all the other values are hashed starting with .NET Core 2.1.100 SDK.
 
+### Template Engine telemetry
```
Comment (tdykstra) [inline docs/core/tools/telemetry.md:118]:
```suggestion
### Template engine telemetry
```

File: docs/core/tools/telemetry.md [hunk 2]
```diff
@@ -115,6 +115,13 @@ A subset of commands sends selected options if they're used, along with their va
...
+
+In addition to the options listed above, the `dotnet new` template instantiation command collects additional data for Microsoft-authored templates (after 2.1.100):
```
Comment (tdykstra) [inline docs/core/tools/telemetry.md:120]:
```suggestion
The `dotnet new` template instantiation command collects additional data for Microsoft-authored templates, starting with .NET Core 2.1.100 SDK:
```

**Review by tdykstra — APPROVED** [review]
Looks good, I left a couple of suggestions.

**Comment by baronfel** [comment]
Thanks for the suggestions, @tdykstra, I took them in their entirety.

**Pull request merged**